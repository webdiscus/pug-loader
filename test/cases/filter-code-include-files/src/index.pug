mixin code
  pre.language-: code
    block

doctype html
html
  head
    link(rel='stylesheet' href=require('./styles.scss'))
    script(src=require('./main.js'))
  body
    //- Manual simulate errors:
    //-div= undefined.variable
    //-img(src=require('NoFound/image.jpeg2'))
    //-
      // alias `@data` from tsconfig
      var cat = 'cat-1';
      var colors = require(`@data/${cat}/color.js`)

    .content
      h1 Watch of dependencies
      +code: :highlight(pug)
        p Pug code
      +code: include:highlight(pug) deps/file.pug
      +code: include:highlight(js) deps/file.js
      +code: include:highlight(js) deps/file.mjs
      +code: include:highlight(js) deps/file.jsx
      +code: include:highlight(js) deps/file.ts
      +code: include:highlight(js) deps/file.tsx
      +code: include:highlight(js) deps/file.json
      //- watch files added via the `watchFiles` option
      +code: include:highlight(js) deps/main.c
      +code: include:highlight(css) deps/file.css
      +code: include:highlight(scss) deps/file.scss
      +code: include:highlight(scss) styles.scss

      .markdown-body
        include:markdown deps/markdown.md
