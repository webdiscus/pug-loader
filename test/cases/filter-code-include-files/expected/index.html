<!DOCTYPE html><html><head><link rel="stylesheet" href="styles.css"><script src="main.js"></script></head><body><div class="content"><h1>Watch of dependencies</h1><pre class="language-"><code><span class="token tag">p</span> <span class="token plain-text">Pug code</span></code></pre><pre class="language-"><code><span class="token tag">p</span> <span class="token plain-text">Hello Pug!</span></code></pre><pre class="language-"><code>console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'file.js'</span><span class="token punctuation">)</span><span class="token punctuation">;</span>
</code></pre><pre class="language-"><code>console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'file.mjs'</span><span class="token punctuation">)</span><span class="token punctuation">;</span></code></pre><pre class="language-"><code>console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'file.jsx'</span><span class="token punctuation">)</span><span class="token punctuation">;</span>
</code></pre><pre class="language-"><code>console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'file.ts'</span><span class="token punctuation">)</span><span class="token punctuation">;</span></code></pre><pre class="language-"><code>console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'file.tsx'</span><span class="token punctuation">)</span><span class="token punctuation">;</span>
</code></pre><pre class="language-"><code><span class="token punctuation">{</span>
  <span class="token string-property property">"type"</span><span class="token operator">:</span> <span class="token string">"JSON"</span>
<span class="token punctuation">}</span></code></pre><pre class="language-"><code><span class="token comment">// main.c</span>
#include <span class="token operator">&lt;</span>stdio<span class="token punctuation">.</span>h<span class="token operator">></span>
#include <span class="token operator">&lt;</span>stdlib<span class="token punctuation">.</span>h<span class="token operator">></span>
</code></pre><pre class="language-"><code><span class="token comment">/* CSS */</span>
<span class="token selector">h1</span> <span class="token punctuation">{</span>
    <span class="token property">color</span><span class="token punctuation">:</span> red<span class="token punctuation">;</span>
<span class="token punctuation">}</span></code></pre><pre class="language-"><code><span class="token comment">// SCSS</span>
<span class="token selector">h1 </span><span class="token punctuation">{</span>
  <span class="token property">color</span><span class="token punctuation">:</span> #C21F39<span class="token punctuation">;</span>
<span class="token punctuation">}</span></code></pre><pre class="language-"><code><span class="token comment">// styles.scss</span>
<span class="token comment">//@use 'github-markdown-css';</span>
<span class="token keyword">@use</span> <span class="token string">'prismjs'</span><span class="token punctuation">;</span>

<span class="token selector">body </span><span class="token punctuation">{</span>
  <span class="token property">font-family</span><span class="token punctuation">:</span> sans-serif<span class="token punctuation">;</span>
  <span class="token property">font-size</span><span class="token punctuation">:</span> 14px<span class="token punctuation">;</span>
  <span class="token property">margin</span><span class="token punctuation">:</span> 0<span class="token punctuation">;</span>
<span class="token punctuation">}</span>

<span class="token selector">h1 </span><span class="token punctuation">{</span>
  <span class="token property">color</span><span class="token punctuation">:</span> darkcyan<span class="token punctuation">;</span>
<span class="token punctuation">}</span>

<span class="token selector">.content </span><span class="token punctuation">{</span>
  <span class="token property">margin</span><span class="token punctuation">:</span> 32px<span class="token punctuation">;</span>
<span class="token punctuation">}</span>

<span class="token selector">.container </span><span class="token punctuation">{</span>
  <span class="token property">margin-top</span><span class="token punctuation">:</span> 32px<span class="token punctuation">;</span>
  <span class="token property">margin-bottom</span><span class="token punctuation">:</span> 32px<span class="token punctuation">;</span>
<span class="token punctuation">}</span>
</code></pre><div class="markdown-body"><h1>Markdown</h1>
<p>File: <strong>markdown.md</strong></p>
<p><em>Italic</em></p>
<p><strong>Bold</strong></p>
<h2>H2 markdown</h2>
<h2>H2 HTML tag</h2>
<pre class="language-html"><code>Highlight HTML code in markdown: <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>span</span><span class="token punctuation">></span></span>span tag<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>span</span><span class="token punctuation">></span></span>
</code></pre>
<blockquote>
<p><strong>Note</strong></p>
<p>This is a note.</p>
</blockquote>
<blockquote>
<p><strong>Warning</strong></p>
<p>This is a warning.</p>
</blockquote>
</div></div></body></html>